lib/handler/redirect-handler.js

```diff
@@ -132,10 +132,20 @@ class RedirectHandler {
     this.opts.maxRedirections = 0
     this.opts.query = null
 
+    // https://tools.ietf.org/html/rfc7231#section-6.4.2
+    // https://fetch.spec.whatwg.org/#http-redirect-fetch
+    // In case of HTTP 301 or 302 with POST, change the method to GET
+    if ((statusCode === 301 || statusCode === 302) && this.opts.method === 'POST') {
+      this.opts.method = 'GET'
+      util.destroy(this.opts.body)
```

Comment:
```suggestion
      util.destroy(this.opts.body.on('error', () => {})
```

Reply:
Maybe is better to use `util.isStream()`?